packages/frontend/src/utils/csvUtils.ts

```diff
@@ -0,0 +1,39 @@
+import { CreateSavedChartVersion } from '@lightdash/common';
+
+export const filterFieldsNotVisible = (
+    chart: CreateSavedChartVersion,
+    columnOrder: string[],
+) => {
+    const { metricQuery } = chart;
+    if (
+        chart.chartConfig.type === 'table' &&
+        chart.chartConfig.config?.columns &&
+        Object.keys(chart.chartConfig.config?.columns).length > 0
+    ) {
+        const filteredColumnIds = Object.entries(
+            chart.chartConfig.config?.columns,
+        ).reduce<string[]>((acc, [columnId, columnConfig]) => {
+            if (columnConfig.visible === false) return [...acc, columnId];
+            return acc;
+        }, []);
+
+        const filterFieldNotVisible = (fieldId: string) =>
+            !filteredColumnIds?.includes(fieldId);
+
+        return {
+            metricQuery: {
+                ...metricQuery,
+                metrics: metricQuery.metrics.filter(filterFieldNotVisible),
+                dimensions: metricQuery.dimensions.filter(
+                    filterFieldNotVisible,
+                ),
```

Comment:
I don't think we should manipulate the metricQuery values.
What happens if a table calculation depends on the dimension/metric?

Reply:
yeah, great point, instead we should pass the table config and hide the columns on the csv generation, right ? 


I can simply remove the filtering to fix https://github.com/lightdash/lightdash/issues/7753 but not https://github.com/lightdash/lightdash/issues/7702